### PR TITLE
Add `FAQ` block

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR adds the FAQ block to the block protocol.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204000740778935/1204065050320022/f) _(internal)_

## 🔍 What does this change?

- Copied How-To block and reskinned it to look like the designs for the FAQ block;
- Made small fixes to `EditableField` and `GetHelpLink` in the design system.

## 📹 Demo

demo

### CODE REVIEW

**Comment by nonparibus** [comment]
(Auto-merge was a fat thumb)

Looking like linting still needs addressing

**Comment by nonparibus** [comment]
@luisbettencourt @hashintel/frontend Please tag me for legal review on Slack when this is FE reviewed. Let's aim to get it in ASAP.

File: blocks/faq/src/question.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+              : {}),
+          }}
+        >
+          <EditableField
+            editIconFontSize={15}
+            value={questionValue}
+            onChange={(event) => {
+              if (!readonly) {
+                setQuestionValue(event.target.value);
+              }
```
Comment (yusufkinatas) [inline blocks/faq/src/question.tsx:110]: Looks like `readonly` prop is already passed to this component down below. What is the benefit of using this if statement here?

File: blocks/faq/src/question.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            }}
+            onBlur={(event) => updateField(event.target.value, questionKey)}
+            sx={{
+              width: 1,
```
Comment (yusufkinatas) [inline blocks/faq/src/question.tsx:114]: On the default style of `EditableField`, it already has `width: 1` if I'm not mistaken. We can remove this if that's the case (please confirm before doing so)

File: blocks/faq/src/app.tsx [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+    );
+  };
+
+  useEffect(() => {
+    if (!questionEntities.length) {
+      void addQuestion();
+    }
+    // We only want to run this once when the block is initiated
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (yusufkinatas) [inline blocks/faq/src/app.tsx:219]: Do we also want to run this on `readonly` mode?. 
This is probably an edge case, but what should happen if we render an empty question list in `readonly` mode?

File: blocks/faq/src/app.tsx [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+          "@type": "Question",
+          name: schemaQuestion,
+          acceptedAnswer: {
+            "@type": "Answer",
+            text: schemaAnswer ?? schemaQuestion,
+          },
+        }),
+      ),
+    });
+  }, [questionEntities]);
```
Comment (yusufkinatas) [inline blocks/faq/src/app.tsx:245]: You probably already did this, but asking to make sure. Is this data tested using a tool like https://developers.google.com/search/docs/appearance/structured-data to make sure it's correct?

File: blocks/faq/src/app.tsx [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+                    paddingY: 3,
+                    paddingX: 3.75,
+                  }
+                : {
+                    background: "none",
+                    boxShadow: "none",
+                  }),
+            }}
+          >
+            {title || description || !readonly ? (
```
Comment (yusufkinatas) [inline blocks/faq/src/app.tsx:346]: Maybe this could be a variable with a more self-explanatory name like `shouldDisplayXYZ`

**Review by yusufkinatas — CHANGES_REQUESTED** [review]
Great work Luis! Dropped few comments & reminders.

File: blocks/faq/src/question.tsx (hunk 1 above)
Comment (luisbettencourt) [reply blocks/faq/src/question.tsx]: Removed in a762683d0d6ba912a6d85128098d6530163102fd, thanks!

File: blocks/faq/src/app.tsx (hunk 4 above)
Comment (luisbettencourt) [reply blocks/faq/src/app.tsx:245]: Yeah I've tested this before but for reference (in portuguese, sorry!):
image

File: blocks/faq/src/app.tsx (hunk 5 above)
Comment (luisbettencourt) [reply blocks/faq/src/app.tsx]: Done in 8a5e5ad9cbd73d52e7dd235d08d964c40b2b2d39

File: blocks/faq/public/book-arrow-right.svg
[the remark this answers is not in the document]
Comment (luisbettencourt) [reply blocks/faq/public/book-arrow-right.svg:1]: Oops, had forgotten this. Updated it in b54e510d067d20960d25732ea2c4ca7c4db24f4f. Thank you for noticing!

File: blocks/faq/src/dev.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      blockInfo={packageJson.blockprotocol}
+      debug
+      initialData={{
+        initialEntities: [testEntity],
+      }}
+      simulateDatastoreLatency={{
+        // configure this to adjust the range of artificial latency in responses to datastore-related requests (in ms)
+        min: 50,
+        max: 200,
+      }}
```
[the remark this answers is not in the document]
Comment (yusufkinatas) [reply blocks/faq/src/dev.tsx:36]: Oooh, my bad for not noticing it.

File: blocks/faq/src/types/block-entity.ts
[the remark this answers is not in the document]
Comment (luisbettencourt) [reply blocks/faq/src/types/block-entity.ts:1]: Done in 0b6a117ad51449e81b821f31b0259641349b3133

**Review by yusufkinatas — APPROVED** [review]
LGTM! 
@nonparibus your turn

**Review by nonparibus — APPROVED** [review]

**Pull request merged**